src/psyclone/psyir/tools/dependency_tools.py

```diff
@@ -273,102 +224,443 @@ def _is_loop_suitable_for_parallel(self, loop, only_nested_loops=True):
             all_loops = loop.walk(Loop)
             if len(all_loops) == 1:
-                self._add_info("Not a nested loop.")
+                self._add_message("Not a nested loop.",
+                                  DACode.INFO_NOT_NESTED_LOOP)
                 return False
 
         if self._loop_types_to_parallelise:
             if loop.loop_type not in self._loop_types_to_parallelise:
-                self._add_info(f"Loop has wrong loop type '{loop.loop_type}'.")
+                self._add_message(f"Loop has wrong loop type '"
+                                  f"{loop.loop_type}'.",
+                                  DACode.INFO_WRONG_LOOP_TYPE)
                 return False
         return True
 
     # -------------------------------------------------------------------------
-    def array_access_parallelisable(self, loop_variable, var_info):
-        '''Tries to determine if the access pattern for a variable
-        given in `var_info` allows parallelisation along the variable
-        `loop_variable`. The following messages might be provided
-        to the user using the message API:
+    @staticmethod
+    def get_flat_indices(component_indices, set_of_loop_vars):
+        '''This function takes an array reference, and returns a flat
+        list of which variable from the given set of variables is used in each
+        subscript. For example, `a(i1+i2)%b(j*j+j,k)%c(l,5)` would return
+        `[(i1,i2),(j,k),(l),()]`. This result is used in partitioning
+        subscripts in the dependency analysis.
+
+        :param component_indices: the component indices used in an array \
+            access.
+        :type component_indices:  \
+            :py:class:`psyclone.core.component_indices.ComponentIndices`
+        :param set_of_loop_vars: set with name of all loop variables.
+        :type set_of_loop_vars: set of str
+
+        :return: a list of sets with all variables used in the corresponding \
+            array subscripts as strings.
+        :rtype: list of sets of str
 
-        * if the array access does not depend on the loop variable, a
-          warning is added (e.g. for the variable `a` in `a(1,2) = b(i,j)`).
-        * if the array variable is accessed inconsistently, e.g.
-          `a(i,j) = a(j,i) + 1`.
+        '''
+        indices = []
+        for i in component_indices.iterate():
+            indx = component_indices[i]
+            index_vars = VariablesAccessInfo(indx)
+            unique_vars = set(str(sig) for sig in index_vars.keys())
+            unique_vars = unique_vars.intersection(set_of_loop_vars)
+            indices.append(unique_vars)
+        return indices
 
-        :param loop_variable: symbol of the variable that is parallelised.
-        :type loop_variable: :py:class:`psyclone.psyir.symbol.DataSymbol`
+    # -------------------------------------------------------------------------
+    @staticmethod
+    def partition(comp_ind1, comp_ind2, loop_vars):
```

Comment:
loop_vars docstring is missing. And I would probably drop the  "loop_" part.

Reply:
Done.